frontend/src/lib/k8s/index.ts

```diff
@@ -173,6 +173,86 @@ export function useConnectApi(...apiCalls: (() => CancellablePromise)[]) {
   );
 }
 
+// Label selector examples: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#list-and-watch-filtering
+// deployment selector example: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements
+// Possible operators: https://github.com/kubernetes/apimachinery/blob/be3a79b26814a8d7637d70f4d434a4626ee1c1e7/pkg/selection/operator.go#L24
+// Format rule for expressions: https://github.com/kubernetes/apimachinery/blob/be3a79b26814a8d7637d70f4d434a4626ee1c1e7/pkg/labels/selector.go#L305
+export function labelSelectorToQuery(labelSelector: LabelSelector) {
+  const segments: string[] = [];
+
+  const matchLabels = labelSelector.matchLabels ?? {};
+  for (const k in matchLabels) {
+    segments.push(`${k}=${matchLabels[k]}`);
+  }
+
+  const matchExpressions = labelSelector.matchExpressions ?? [];
+  for (const expr of matchExpressions) {
+    let segment = '';
+    if (expr.operator === 'DoesNotExist') {
+      segment += '!';
+    }
+
+    let needsParensWrap = false;
+    let expectedValuesLength = -1; // -1 means no limit
```

Comment:
I was confused reading the code below where it's used about what -1 meant. Eventually I found the comment, but Instead of -1, we could use a string instead and a union type.

Something like...

```ts
expectedValuesLength : string | number = 'no limit'
...
    } else if (expectedValuesLength === 'no limit') {
...
```

Reply:
I don't love the type overload for just interpreting a number, so I added a const that represents the no-limits.